sklearn/metrics/pairwise.py

```diff
@@ -1835,15 +1839,21 @@ def _pairwise_callable(X, Y, metric, force_all_finite=True, **kwds):
         # Calculate diagonal
         # NB: nonzero diagonals are allowed for both metrics and kernels
         for i in range(X.shape[0]):
-            x = X[i]
+            # scipy not yet implemented 1D sparse slices; can be removed and use simply
+            # `arr[ind]` once implemented
+            x = X[[i], :] if issparse(X) else X[i]
             out[i, i] = metric(x, x, **kwds)
 
     else:
         # Calculate all cells
         out = np.empty((X.shape[0], Y.shape[0]), dtype="float")
         iterator = itertools.product(range(X.shape[0]), range(Y.shape[0]))
         for i, j in iterator:
-            out[i, j] = metric(X[i], Y[j], **kwds)
+            # scipy not yet implemented 1D sparse slices; can be removed and use simply
+            # `arr[ind]` once implemented
+            x = X[[i], :] if issparse(X) else X[i]
+            y = Y[[j], :] if issparse(Y) else Y[j]
+            out[i, j] = metric(x, y, **kwds)
```

Comment:
We will need an entry to acknowledge which public functionality will be impacted by this change.

Reply:
Seems that there are a lot of public interfaces that will be affected, so I may need some time to find them all (hopefully).